### ISSUE TITLE
Getting an error when running on kubernetes with persistent volume (OperationalError: no such table: auth_user)

### ISSUE THREAD

**Issue by groupvrg** [graphite-project/docker-graphite-statsd#46]
Hi,

*** I think I found the problem, the persistent volume is created with a lost+found dir, this cause:
this variable:
`graphite_storage_dir_contents=$(find /opt/graphite/storage -mindepth 1 -print -quit)`
 in [01_conf_init.sh](https://github.com/graphite-project/docker-graphite-statsd/blob/master/conf/etc/my_init.d/01_conf_init.sh) to not be empty, because of this django_admin_init.exp is not invoked:
```
if [[ -z $graphite_storage_dir_contents ]]; then
  /usr/local/bin/django_admin_init.exp
fi
```

------
original:
I'm trying to run the docker image on kubernetes when I try to enter the portal I get this error:

>Traceback (most recent call last):
>  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/exception.py", line 41, in inner
>    response = get_response(request)
>  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 187, in _get_response
>    response = self.process_exception_by_middleware(e, request)
>  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 185, in _get_response
>    response = wrapped_callback(request, *callback_args, **callback_kwargs)
>  File "/opt/graphite/webapp/graphite/composer/views.py", line 35, in composer
>    profile = getProfile(request)
>  File "/opt/graphite/webapp/graphite/user_util.py", line 25, in getProfile
>    return default_profile()
>  File "/opt/graphite/webapp/graphite/user_util.py", line 41, in default_profile
>    'password': '!'})
>   File "/usr/local/lib/python2.7/dist-packages/django/db/models/manager.py", line 85, in manager_method
>     return getattr(self.get_queryset(), name)(*args, **kwargs)
>   File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 464, in get_or_create
>     return self.get(**lookup), False
>   File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 374, in get
>     num = len(clone)
>   File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 232, in __len__
>     self._fetch_all()
>   File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 1118, in _fetch_all
>     self._result_cache = list(self._iterable_class(self))
>   File "/usr/local/lib/python2.7/dist-packages/django/db/models/query.py", line 53, in __iter__
>     results = compiler.execute_sql(chunked_fetch=self.chunked_fetch)
>   File "/usr/local/lib/python2.7/dist-packages/django/db/models/sql/compiler.py", line 900, in execute_sql
>     raise original_exception
> OperationalError: no such table: auth_user

this is the configuration I'm using:

```
 ---
apiVersion: v1
kind: Service
metadata:
  name: graphite-svc
  labels:
    app: graphite
spec:
  type: LoadBalancer
  ports:
  - port: 80
    targetPort: 80
  selector:
    app: graphite
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: graphite-persistent-storage
spec:
  accessModes:
    - ReadWriteOnce
  storageClassName: fast
  resources:
    requests:
      storage: 300Gi
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: graphite-deploy
  labels:
    app: graphite
spec:
  replicas: 1
  selector:
    matchLabels:
      app: graphite
  template:
    metadata:
      labels:
        app: graphite
    spec:
      containers:
      - name: graphite
        image: graphiteapp/graphite-statsd
        ports:
         - containerPort: 80
         - containerPort: 2003
           protocol: UDP
        volumeMounts:
          - name: graphite-data
            mountPath: /opt/graphite/storage
        env:
          - name: GRAPHITE_STORAGE_DIR
            value: /opt/graphite/storage
      volumes:
      - name: graphite-data
        persistentVolumeClaim:
          claimName: graphite-persistent-storage 
```

when running without persistent volume the image works as expected. I don't really know where to start looking. the only other different thing I see is that there is a lost+found dir in /opt/graphite/storage

Thanks.

**Comment by deniszh** [comment]
Yes, init script should be fixed to skip `lost+found` dir

**Comment by deniszh** [comment]
Fixed in 1.1.4-1 release

**Comment by ertwroc** [comment]
Please reopen this issue - now graphite doesn't start when you run it second time. It hangs on `django-admin.py createsuperuser`

```
root@graphite:/# ps axf
  PID TTY      STAT   TIME COMMAND
   44 ?        Ss     0:00 /bin/sh
   51 ?        S      0:00  \_ bash
  100 ?        R+     0:00      \_ ps axf
    1 ?        Ss     0:00 /usr/bin/python3 -u /sbin/my_init
    9 ?        S      0:00 /bin/bash /etc/my_init.d/01_conf_init.sh
   19 ?        Sl     0:00  \_ expect /usr/local/bin/django_admin_init.exp
   20 pts/0    Ss+    0:00      \_ /bin/bash /usr/local/bin/manage.sh
   39 pts/0    S+     0:00          \_ /usr/bin/python /usr/local/bin/django-admin.py createsuperuser --settings=graphite.settings
```

The problem is with line `graphite_storage_dir_contents=$(find /opt/graphite/storage -mindepth 1 -print -quit | grep -v lost+found)` in `/etc/my_init.d/01_conf_init.sh`. When `find` returns as first and the only line `lost+found` then it is excluded by `grep` no mater if there are any other directories. It can be solved by doing exclude directly in `find` command, so please replace this line with

```
graphite_storage_dir_contents=$(find /opt/graphite/storage -mindepth 1 -not -path /opt/graphite/storage/lost+found -print -quit | grep -v lost+found)
```

**Comment by deniszh** [comment]
Argh, will fix it asap

**Comment by deniszh** [comment]
Fix released in 1.1.4-3, also opened PR for official helm chart to upgrade.
Thanks for testing, @ertwroc, sorry for messing that up. 😭
